unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarButton.cs

```diff
@@ -26,6 +29,26 @@ public void Initialize()
 
         if (iconImage != null)
             originalIconColor = iconImage.color;
+
+        DataStore.i.appMode.OnChange += AppMode_OnChange;
+        AppMode_OnChange(DataStore.i.appMode.Get(), AppMode.DEFAULT);
+    }
+
+    private void OnDestroy() { DataStore.i.appMode.OnChange -= AppMode_OnChange; }
+
+    private void AppMode_OnChange(AppMode current, AppMode previous)
+    {
+        bool isInteractable = true;
+        foreach (var mode in modesWhereNotInteractable)
+        {
+            if (current == mode)
+            {
+                isInteractable = false;
+                break;
+            }
+        }
```

Comment:
Is not better to implement the counterpart logic here? You should specify in which modes it should work instead of in which modes it doesn't work.

This way in the future if you are developing a new feature you only need to take care about the things that needs to work and not about the things that souldn't work.

In the future if we have like 34 systems they will probably work independent of each other, so it is more scalable the other way

Maybe is only my opinion!

Reply:
I like this suggestion, It could minimize a lot of possible issues in the future

Reply:
Yeah, I guess it can be interpreted in both ways. For me the fact of disable a feature is always the exception, in fact in the case of the taskbar we only need to disable some buttons in a very specific case (when the BiW editor is open) and the rest of situations should be actived. But I understand what you say... I don't know, I think we can let each consumer decide how to handle it.

Reply:
Well if you both agree, I'm going to change it 👍

Reply:
Yeah, the main reason for me about this could be that if you are implementing a new feature, it would be easy to notice that something is not working when it should than noticing that something is working when it shouldn't :P

Reply:
But take into account that, with your approach, when we add any new AppMode, we are going to need to go to each system (taskbar and anye others that is using the enum) and add the new mode in its "list of compatible modes".

Reply:
Done in https://github.com/decentraland/unity-renderer/pull/333/commits/52af5e303ad1ed2f17db2974b2f73db3e6df7359. Thanks guys!